internal/expr/parser.go

```diff
@@ -119,6 +174,11 @@ func (p *Parser) Parse(input string) (expr *ParsedExpr, err error) {
 	}()
 
 	p.init(input)
+
+	if err := p.removeComments(); err != nil {
```

Comment:
should this call be placed inside `p.init`? This would avoid the second call to `p.init` made inside `removeCommants` (so line 163 could be replaced with just `p.input = res.String()`)

Reply:
The problem with that is more the error handling, I think its good that p.init doesn't return an error, it doens't feel like it should, and we should probably be aware we are removing the comments.